includes/class-sensei-grading.php

```diff
@@ -1147,7 +1170,8 @@ public static function get_user_graded_lessons_sum( $user_id ) {
 		$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
 		$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade') AND {$wpdb->comments}.user_id = {$clean_user_id} ";
 
-		$comment_query     = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+		$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+		// WPCS: unprepared SQL OK.
```

Comment:
Could we specify why it's OK?

Reply:
Soo, I didn't touch this code, I just saw a linter error for this code. I had some general issues with the linter so I wanted to minimize the warnings, but I can remove the comment.

Reply:
While that may be true for the existing code examples, we're also compounding the problem by copying it and using the same approach again [here](https://github.com/Automattic/sensei/pull/4865/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1142). The linter is warning us that there's potentially a serious security problem here, so instead of suppressing the warning it would be better to fix the root cause. There are a few examples of how to do this in the codebase. [Here's one.](https://github.com/Automattic/sensei/blob/df562dded9cf3ae9f5cc3ec341adfe9496aea1eb/includes/class-sensei-data-cleaner.php#L384)

Reply:
I agree. You could inline the query and use `$wpdb->prepare()` to pass the variable. It's a somewhat standard way to do custom queries.

Reply:
@miro and @donnapep  so regarding this issue. 

1. I removed the comment from the functions that I didn't touch for this PR and Task - I didn't want to leave the comment but also I didn't want to change the functions (query) and so on. I don't want something to break.

2. Query I have doesn't need any arguments so it is not practiced to do the preparation when we don't have arguments. But because of this discussion, I decided to use it anyway. First I tried calling ```$wpdb->prepare()``` with a single argument but it is not working like that, then I've added an argument 'grade' just to have something and it worked. 

3. Interesting enough even when I've added ```$wpdb->prepare()```  the linter error was still there so I surpassed it with the comment. Here is the error in github actions: 
<img width="794" alt="image" src="https://user-images.githubusercontent.com/7208249/157231570-9d41978b-6918-4b4e-9380-83602f24f169.png">

4. Since it belongs to the same part of the code. I had a linter issue about using ```wp_cace_add/_set ``` I actually liked the idea since this query is kind of heavy and the data stored is just a single value.  I figured it is enough to have it refreshed on the new page load. Do you agree with me? 

5. Linter issue about calling directly DB i had to put comment for ignoring that: 
<img width="792" alt="image" src="https://user-images.githubusercontent.com/7208249/157231885-5e020765-672f-4b8b-ac27-c5c7c7cb6ab0.png">

Reply:
I just submitted a [draft PR](https://github.com/Automattic/sensei/pull/4904) and pushed a [commit](https://github.com/Automattic/sensei/pull/4904/commits/542c7437bcf06453a547069ac959494ddb81efcb) so you can see what I did to address the PHPCS errors and warnings.

Some other things worth noting about that commit:
- [I've escaped the output](https://github.com/Automattic/sensei/pull/4904/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR62). Even though I know the output is safe, it's considered a WP best practice to escape late like this.
- I've opted [not to make this function `static`](https://github.com/Automattic/sensei/pull/4904/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1213) like the other ones in this class. I don't have a strong reason for doing this other than the fact that it would be harder to move from a static to a non-static method if we discovered that we suddenly needed access to some of the class variables in the future for some reason.
- As per our discussion in the team meeting, we don't need to cache so I'm [ignoring the linter warning](https://github.com/Automattic/sensei/pull/4904/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1216). Same thing for the direct query warning, since we've identified it as a performance improvement. But I've added a brief reason to denote why I'm ignoring PHPCS, to show that I've given some thought to it.
- I don't think you need to use `wpdb->prepare` at all. The only reason it's needed now is because of the [placeholder variable for the `meta_key`](https://github.com/Automattic/sensei/pull/4865/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1147). But that's actually not a variable, but a string (i.e. `grade`), so that can just be part of the query. You can probably use the exact same approach as me for your query and swap out what you need.

Reply:
1. Added [escaping output](https://github.com/Automattic/sensei/pull/4865/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR77)
2. Made function [not static ](https://github.com/Automattic/sensei/pull/4865/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1141) so that we have consistency through the repo
3. I removed the [caching](https://github.com/Automattic/sensei/pull/4865/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1151)
4. And I removed the [preparig](https://github.com/Automattic/sensei/pull/4865/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1151)